test/relayHub/RelayHub.test.ts

```diff
@@ -0,0 +1,1199 @@
+                    await TestDeployVerifierEverythingAccepted.new();
+                gaslessAccount = await getGaslessAccount();
+
+                const smartWalletTemplate: SmartWalletInstance =
+                    await SmartWallet.new();
+                factory = await createSmartWalletFactory(smartWalletTemplate);
+                recipientContract = await TestRecipient.new();
+
+                const testToken = artifacts.require('TestToken');
+                token = await testToken.new();
+
+                target = recipientContract.address;
+                verifier = verifierContract.address;
+                relayHub = relayHubInstance.address;
+
+                sharedDeployRequestData = {
+                    request: {
+                        relayHub: relayHub,
+                        to: constants.ZERO_ADDRESS,
+                        data: '0x',
+                        from: gaslessAccount.address,
+                        nonce: (
+                            await factory.nonce(gaslessAccount.address)
+                        ).toString(),
+                        value: '0',
+                        tokenContract: token.address,
+                        tokenAmount: '1',
+                        tokenGas: '50000',
+                        recoverer: constants.ZERO_ADDRESS,
+                        index: '0'
+                    },
+                    relayData: {
+                        gasPrice,
+                        relayWorker,
+                        callForwarder: factory.address,
+                        callVerifier: deployVerifierContract.address,
+                        domainSeparator: getDomainSeparatorHash(
+                            factory.address,
+                            chainId
+                        )
+                    }
+                };
+
+                deployRequest = cloneDeployRequest(sharedDeployRequestData);
+                deployRequest.request.index = nextWalletIndex.toString();
+                misbehavingVerifier =
+                    await TestDeployVerifierConfigurableMisbehavior.new();
+                deployRequest.request.index = nextWalletIndex.toString();
+                nextWalletIndex++;
+
+                relayRequestMisbehavingVerifier =
+                    cloneDeployRequest(deployRequest);
+                relayRequestMisbehavingVerifier.relayData.callVerifier =
+                    misbehavingVerifier.address;
+
+                const dataToSign = new TypedDeployRequestData(
+                    chainId,
+                    factory.address,
+                    relayRequestMisbehavingVerifier
+                );
+                signatureWithMisbehavingVerifier = getLocalEip712Signature(
+                    dataToSign,
+                    gaslessAccount.privateKey
+                );
+            });
+
+            it('Should faild a deployRequest if SmartWallet has already been initialized', async () => {
```

Comment:
misspelled

Reply:
The word faild is misspelled and should be updated to failed.

Reply:
✅